### PR TITLE
Add provider e2e tests

### PR DESCRIPTION
**Pull request by hydrogen18**
Changes

1. Add provider e2e tests
2. Fix some issues I found around service-status & lease-status having missing data
3. Refactor the common function I had to use the cosmos SDK context so the tests can capture
the output & test against it

### CODE REVIEW

File: x/market/client/cli/lease.go [hunk 1]
```diff
@@ -59,7 +59,9 @@ func cmdGetLease() *cobra.Command {
 		Short: "Query order",
 		Args:  cobra.ExactArgs(0),
 		RunE: func(cmd *cobra.Command, args []string) error {
+
 			clientCtx := client.GetClientContextFromCmd(cmd)
+			fmt.Printf("clientCtx: %+v\n", clientCtx)
```
Comment (anilCSE) [inline x/market/client/cli/lease.go:64]:
```suggestion
```

File: x/market/client/cli/lease.go [hunk 2]
```diff
@@ -71,7 +73,7 @@ func cmdGetLease() *cobra.Command {
 			if err != nil {
 				return err
 			}
-
+			fmt.Printf("bidID:%+v\n", bidID)
```
Comment (anilCSE) [inline x/market/client/cli/lease.go:76]:
```suggestion
```

File: cmd/common/util_test.go [hunk 3]
```diff
@@ -1,29 +1,33 @@
 package common
 
 import (
+	"github.com/cosmos/cosmos-sdk/client"
```
Comment (anilCSE) [inline cmd/common/util_test.go:4]: sort imports?

File: cmd/common/util_test.go [hunk 4]
```diff
@@ -1,29 +1,33 @@
...
 )
 
 func TestPrintJSONStdoutStruct(t *testing.T) {
+	ctx := client.Context{}
 	var x struct{ foo int }
 	x.foo = 55
-	err := PrintJSONStdout(x)
+	err := PrintJSON(ctx, x)
 	if err != nil {
 		t.Errorf("PrintJSONStdout failed:[%T] %v", err, err)
```
Comment (anilCSE) [inline cmd/common/util_test.go:14]:
```suggestion
		t.Errorf("PrintJSON failed:[%T] %v", err, err)
```

File: cmd/common/util_test.go [hunk 5]
```diff
@@ -1,29 +1,33 @@
...
 	}
 }
 
 func TestPrintJSONStdoutInt(t *testing.T) {
+	ctx := client.Context{}
 	x := 123
-	err := PrintJSONStdout(x)
+	err := PrintJSON(ctx, x)
 	if err != nil {
 		t.Errorf("PrintJSONStdout failed:[%T] %v", err, err)
```
Comment (anilCSE) [inline cmd/common/util_test.go:23]:
```suggestion
		t.Errorf("PrintJSON failed:[%T] %v", err, err)
```

File: cmd/common/util_test.go [hunk 6]
```diff
@@ -1,29 +1,33 @@
...
 	}
 }
 
 func TestPrintJSONStdoutNil(t *testing.T) {
+	ctx := client.Context{}
 	var x interface{} // implicitly nil
-	err := PrintJSONStdout(x)
+	err := PrintJSON(ctx, x)
 	if err != nil {
 		t.Errorf("PrintJSONStdout failed:[%T] %v", err, err)
```
Comment (anilCSE) [inline cmd/common/util_test.go:32]:
```suggestion
		t.Errorf("PrintJSON failed:[%T] %v", err, err)
```

File: integration/e2e_test.go [hunk 7]
```diff
@@ -3,7 +3,9 @@
 package integration
 
 import (
+	"encoding/json"
 	"fmt"
+	"github.com/stretchr/testify/assert"
```
Comment (anilCSE) [inline integration/e2e_test.go:8]: sort imports?

**Review by anilCSE — APPROVED** [review]
lgtm, some nits

File: x/market/client/cli/lease.go (hunk 1 above)
Comment (hydrogen18) [reply x/market/client/cli/lease.go:64]: Good catch, I forgot I was debugging this code. I'll remove it.

File: cmd/common/util_test.go (hunk 3 above)
Comment (hydrogen18) [reply cmd/common/util_test.go:4]: I think I did, I'll run `gofmt` again to make sure things are in the order that is used by it.

**Review by boz — APPROVED** [review]
:pray:

**Pull request merged**